wire-ios/Wire-iOS/Sources/UserInterface/Conversation/InputBar/E2EIPrivacyWarningChecker.swift

```diff
@@ -77,8 +77,8 @@ struct E2EIPrivacyWarningChecker {
     static func addPresenter(_ observer: E2EIPrivacyWarningPresenter) -> NSObjectProtocol {
```

Comment:
The returned token is never unregistered as explained in the documentation:

```swift
        if let observationToken {
            notificationCenter.removeObserver(observationToken)
        }
```

Also I would use a `private let notificationCenter = NotificationCenter.default` to make sure we add and remove observer from the same instance.

Reply:
> Return Value
An opaque object to act as the observer. Notification center strongly holds this return value until you remove the observer registration.
https://developer.apple.com/documentation/foundation/notificationcenter/1411723-addobserver

Reply:
what do you mean it's never unregistered, it's up to the caller to unregister it no?

Reply:
> what do you mean it's never unregistered, it's up to the caller to unregister it no?

When using block based API we are responsible for calling `removeObserver` as opposed the selector based API where it's not necessary.

Reply:
alright so suggestion of markus is to use the selector base API

Reply:
Should I make a commit with suggestion on this branch?

Reply:
So I updated the code use our existing `SelfUnregisteringNotificationCenterToken`

Reply:
we're good to go now